packages/react-instantsearch-dom/src/components/VoiceSearch.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import { translatable, Translate } from 'react-instantsearch-core';
+import { createClassNames } from '../core/utils';
+import voiceSearchHelper, {
+  VoiceSearchHelper,
+  VoiceListeningState,
+} from '../lib/voiceSearchHelper';
+const cx = createClassNames('VoiceSearch');
+
+type InnerComponentProps = {
+  status: string;
+  errorCode?: string;
+  isListening: boolean;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  isBrowserSupported: boolean;
+};
+
+type VoiceSearchProps = {
+  searchAsYouSpeak?: boolean;
+
+  refine: (query: string) => void;
+  translate: Translate;
+  buttonComponent?: React.FC<InnerComponentProps>;
+  statusComponent?: React.FC<InnerComponentProps>;
+};
+
+const getDefaultButtonInnerElement = (
+  status: string,
+  errorCode: string | undefined,
+  isListening: boolean
+) => {
+  if (status === 'error' && errorCode === 'not-allowed') {
+    return (
+      <>
```

Comment:
This is the first time we use Fragments in the codebase. According to our React peer dependencies, it should be fine. Can you confirm @samouss?